File: test/parsergenerator/nullableset_test.scala
package pg.parsergenerator

import org.scalatest.FunSpec
import pg.parsergenerator.nullableset.{NullableSet,generateNulls,isNullable}
import pg.{data, token}
import data.sample_language.test_sample_grammar

class nullableset_test extends FunSpec {

  describe("NullableSet test") {
    val nulls = generateNulls(test_sample_grammar)
    it("T is Nullable") {
      assert(isNullable(nulls, "T"))
    }
    it("LIST is Nullable") {
      assert(isNullable(nulls, "LIST"))
    }
    it("HOGE is not Nullable") {
      assert(!isNullable(nulls, "HOGE"))
    }
    it("E is not Nullable") {
      assert(!isNullable(nulls, "E"))
    }
    it("S is not Nullable") {
      assert(!isNullable(nulls, "S"))
    }
  }
}

File: test/parsergenerator/nullableset_test.ml
open OUnit
open Nullableset
open Token
open Sample_language

let test () =
  let nulls = generateNulls(test_sample_grammar) in
  "NullableSet test" >::: [
    "T is Nullable" >:: begin fun () ->
      assert(isNullable(nulls, "T"))
    end;
    "LIST is Nullable" >:: begin fun () ->
      assert(isNullable(nulls, "LIST"))
    end;
    "HOGE is not Nullable" >:: begin fun () ->
      assert(not(isNullable(nulls, "HOGE")))
    end;
    "E is not Nullable" >:: begin fun () ->
      assert(not(isNullable(nulls, "E")))
    end;
    "S is not Nullable" >:: begin fun () ->
      assert(not(isNullable(nulls, "S")))
    end;
  ]
